micromanager_gui/_core_widgets/_snap_button_widget.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional, Tuple, Union
+
+from fonticon_mdi6 import MDI6
+from pymmcore_plus import CMMCorePlus
+from qtpy.QtCore import QSize, Qt
+from qtpy.QtGui import QColor
+from qtpy.QtWidgets import QApplication, QPushButton
+from superqt.fonticon import icon
+
+from micromanager_gui._core import get_core_singleton
+
+
+class SnapButton(QPushButton):
+    """
+    Create a snap QPushButton linked to the pymmcore-plus 'snap()' method.
+    Once the button is clicked, an image is snapped and the pymmcore-plus
+    'imageSnapped' signal is emitted.
+    """
+
+    def __init__(
+        self,
+        mmcore: Optional[CMMCorePlus] = None,
+        button_text: Optional[str] = None,
+        icon_size: Optional[int] = 30,
+        icon_color: Optional[
+            Union[
+                QColor,
+                int,
+                str,
+                Qt.GlobalColor,
+                Tuple[int, int, int, int],
+                Tuple[int, int, int],
+            ]
+        ] = "black",
+    ) -> None:
+
+        super().__init__()
+
+        self._mmc = mmcore or get_core_singleton()
+        self.button_text = button_text
+        self.icon_size = icon_size
+        self.icon_color = icon_color
+
+        self._create_button()
+
+    def _create_button(self):
+        if self.button_text:
+            self.setText(self.button_text)
+        self.setIcon(icon(MDI6.camera_outline, color=self.icon_color))
+        self.setIconSize(QSize(self.icon_size, self.icon_size))
+        self.clicked.connect(self._snap)
+
+    def _snap(self):
+        loaded = self._mmc.getLoadedDevices()
```

Comment:
rather than checking this on every snap, let's just try `_mmc.snap()` and catch an error.  Moreover... can we go one step further and listen for sys config loaded signals and disable the button whenever a camera isn't availalable?

Reply:
Also, if we do do a check like this (it could be moved to init) we should not just check for number of loaded devices but instead use either [`mmc.getLoadedDevicesOfType`](https://micro-manager.org/apidoc/MMCore/latest/class_c_m_m_core.html#a193eb1e3d80e3dabee70f1434a2b1b75) or '[`iter_devices`](https://pymmcore-plus.readthedocs.io/en/latest/api/pymmcore_plus.html#pymmcore_plus.CMMCorePlus.iterDevices)